### PR TITLE
spark conf handler plus AQE optimizations

### LINKED ISSUE #771: Par Spark Conf - Replicate global configs to thread-specific new sessions
when thread-specific spark sessions are created the global conf should be mirrored into the new sparkSession and all global overrides should be combined into a single override location.

### PR DESCRIPTION
**Pull request by GeekSheikh**
closes #771

### CODE REVIEW

**Comment by sriram251-code** [comment]
Screenshot 2023-02-21 at 11 54 32 AM
I guess for generateBaseConfig also we need to set spark(glabalSession=true)

File: src/main/scala/com/databricks/labs/overwatch/env/Database.scala [hunk 1]
```diff
@@ -155,9 +155,9 @@ class Database(config: Config) extends SparkSessionWrapper {
          |""".stripMargin)
     df.write.format("delta").save(dfTempPath)
 
-    spark.conf.set("spark.sql.files.maxPartitionBytes", 1024 * 1024 * 16) // maximize parallelism on re-read and let
+    // maximize parallelism on re-read and let AQE bring it back down
+    spark.conf.set("spark.sql.files.maxPartitionBytes", 1024 * 1024 * 16)
```
Comment (sriram251-code) [inline src/main/scala/com/databricks/labs/overwatch/env/Database.scala:159]: why 16 mb ?

**Review by sriram251-code — CHANGES_REQUESTED** [review]

**Comment by GeekSheikh** [comment]
> Screenshot 2023-02-21 at 11 54 32 AM I guess for generateBaseConfig also we need to set spark(glabalSession=true)

doesn't this happen before `parSessionOn` is enabled thus it should just use global by default, right?

File: src/main/scala/com/databricks/labs/overwatch/env/Database.scala (hunk 1 above)
Comment (GeekSheikh) [reply src/main/scala/com/databricks/labs/overwatch/env/Database.scala:159]: I didn't change this. This is to increase parallelism on the persistAndLoad functions to get more parallelism during load and allow AQE to coalesce the partitions during the subsequent shuffles.

**Review by sriram251-code — APPROVED** [review]

**Pull request merged**